clang/include/clang/Basic/Builtins.td

```diff
@@ -4725,6 +4725,12 @@ def HLSLMad : LangBuiltin<"HLSL_LANG"> {
   let Prototype = "void(...)";
 }
 
+def HLSLNormalize : LangBuiltin<"HLSL_LANG"> {
+  let Spellings = ["__builtin_hlsl_normalize"];
```

Comment:
is this an elementwise case? isnt it  `n = normalize(x)  = n[i] = x[i]/ length(x)`

Reply:
I don't believe normalize is an elementwise operation. 
"elementwise" semantically means "for each element". We have operations like elementwise_log, and elementwise_pow, etc. What they do is apply a specific binary operation to each element in a vector. Applying log to each element of a vector is applying log "elementwise", for each element.
The normalize operation can never be applied to a singular element in a vector, it doesn't make sense to normalize a single vector element. Normalization always applies to the whole vector, although what is actually happening is a division across each element in the vector. 
So I don't think elementwise_normalize makes sense, because it doesn't make sense to normalize a singular vector element, say a float.

Reply:
I guess the point i was trying to make was that for each element in x you are dividing by length. Thats actually the same behavior as many of the log cases except those multiplies are via a constant float value while `1/length(x)` is dynamic.

Reply:
I'm fine with this as is. resolving.